### PR TITLE
Adding comments for FirebaseOptions

### PR DESCRIPTION
**Pull request by egilmorez**
Adding comments for FirebaseOptions so that the reference documentation will be populated.

Thanks!

### CODE REVIEW

File: packages-exp/app-exp/src/public-types.ts [hunk 1]
```diff
@@ -70,16 +70,48 @@ export interface FirebaseApp {
 /**
  * @public
  *
- * Firebase configuration object
+ * Firebase configuration object. Contains a set of parameters required by
+ * services in order to successfully communicate with Firebase server APIs
+ * and to associate client data with your Firebase project and
+ * Firebase application.
```
Comment (hsubox76) [inline packages-exp/app-exp/src/public-types.ts:76]: Should we say somewhere that they can get this entire config from the console? Or something so nobody makes the mistake of thinking they have to think about and populate each field individually.

File: packages-exp/app-exp/src/public-types.ts [hunk 2]
```diff
@@ -70,16 +70,48 @@ export interface FirebaseApp {
...
+   * Unique numerical value used to identify each sender that can send
+   * Firebse Cloud Messaging messages to client apps.
+   */
   messagingSenderId?: string;
+  /**
+   * Unique identifier for the app.
+   */
   appId?: string;
+  /**
+   * An ID automatically created when you enable Analytics in your
```
Comment (hsubox76) [inline packages-exp/app-exp/src/public-types.ts:112]: Should we mention that measurementId is optional in v7.20.0 and up?

File: packages-exp/app-exp/src/public-types.ts (hunk 1 above)
Comment (egilmorez) [reply packages-exp/app-exp/src/public-types.ts]: Good point, done.

File: packages-exp/app-exp/src/public-types.ts (hunk 2 above)
Comment (egilmorez) [reply packages-exp/app-exp/src/public-types.ts:114]: Done.

**Review by hsubox76 — APPROVED** [review]

**Review by Feiyang1 — APPROVED** [review]

**Pull request merged**